datamodel/src/main/java/com/hartwig/serve/datamodel/ServeDatabase.java

```diff
@@ -0,0 +1,24 @@
+package com.hartwig.serve.datamodel;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import org.immutables.value.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Value.Immutable
+@Value.Style(passAnnotations = { NotNull.class, Nullable.class },
+             jdkOnly = true)
+@JsonSerialize(as = ImmutableServeDatabase.class)
+@JsonDeserialize(as = ImmutableServeDatabase.class)
+public abstract class ServeDatabase {
+
+    @NotNull
+    public abstract String version();
+
+    @NotNull
+    public abstract Map<RefGenome, ServeRecord> records();
```

Comment:
Just pointing out I this means we'll parse all the json only to usually (or always?) ignore half of it, and the ServeRecord is a bit fat at approaching 70M. But I can see how its convenient from a data management point of view, so ok.

Reply:
Thanks for bringing this up, I didn't think about this technical stuff (and so also have no opinion on it). Will leave it like this for now, if you say it's ok

Reply:
@kzuberihmf In practice we may get the situation where we get tests with different ref genomes, e.g. if we switch internally to 38 while the external tests done by other labs are on 37 then you need to load both SERVE versions. I don't think we want to add this complexity to outside of the application.

If the size is a problem we could make smart loaders potentially? E.g. if ACTIN-molecular starts it could first take the unique set of all ref genome versions of molecular inputs and then only load the ones it needs? Could also optimize within `ServeRecord` (e.g. trial ingestion only needs the known events). Though sounds like something for later if it turns out to be an actual problem?

Reply:
Agree to leave until it becomes an actual problem.